### PR TITLE
Removed the IPv6 prefix size limit for cluster-cidr

### LINKED ISSUE #50395: IPv6 prefix size is limited to 64 with kube-controller-manager
/kind bug

**What happened**:
With https://github.com/kubernetes/kubernetes/pull/43586, 
the controller manager will accept IPv6 addresses
with the  --cluster-cidr flag.
The prefix length is currently limited to /64 with
maxPrefixLength = 64

**What you expected to happen**:
The question here is if this limit should be removed, and allow a greater
prefix size or if this limit should remain.

**How to reproduce it (as minimally and precisely as possible)**:
giving the kube-controller-manager --cluster-cidr a value of 65 or greater

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration**:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by aanm**
Fixes #50395

/sig network
/area ipv6

cc @rpothier

### CODE REVIEW

**Comment by rpothier** [comment]
Thanks for getting to this, there are other functions in the file that need to be updated also for allowing the larger subnet size, can these get updated also?

File: pkg/controller/node/ipam/cidrset/cidr_set_test.go [hunk 1]
```diff
@@ -130,6 +130,69 @@ func TestIndexToCIDRBlock(t *testing.T) {
 			CIDRBlock:      "2001:db8:85a3:3232::/63",
 			description:    "Index with IPv6",
 		},
+		{
+			clusterCIDRStr: "2001:0db8::/32",
+			subnetMaskSize: 48,
+			index:          0,
+			CIDRBlock:      "2001:db8::/48",
+			description:    "Index with IPv6",
```
Comment (rpothier) [inline pkg/controller/node/ipam/cidrset/cidr_set_test.go]: With the increased number of tests, can the description be a bit different so make it easier to identify a failing test?

**Comment by aanm** [comment]
>Thanks for getting to this, there are other functions in the file that need to be updated also for allowing the larger subnet size, can these get updated also?

I was not aware of that, which ones?

**Comment by rpothier** [comment]
Will getBeginingAndEndIndices() and getIndexForIP() need to be updated for 128 bits?

**Comment by aanm** [comment]
@rpothier Done!

File: pkg/controller/node/ipam/cidrset/cidr_set_test.go (hunk 1 above)
Comment (aanm) [reply pkg/controller/node/ipam/cidrset/cidr_set_test.go]: Done

File: pkg/controller/node/ipam/cidrset/cidr_set.go [hunk 2]
```diff
@@ -74,6 +74,48 @@ func NewCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) *CidrSet {
 	}
 }
 
+// TODO: Replace this with math/bits when upgrading to go 1.9
```
[the remark this answers is not in the document]
Comment (aanm) [reply pkg/controller/node/ipam/cidrset/cidr_set.go]: This was copied from golang's lib

File: pkg/controller/node/ipam/cidrset/cidr_set.go [hunk 3]
```diff
@@ -74,6 +74,48 @@ func NewCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) *CidrSet {
...
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+	0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x08,
+}
+
+// TODO: Replace this with math/bits when upgrading to go 1.9
```
[the remark this answers is not in the document]
Comment (aanm) [reply pkg/controller/node/ipam/cidrset/cidr_set.go]: I've rephrase this. This should be removed, what needs to be replaced is on line 150

File: pkg/controller/node/ipam/cidrset/cidr_set.go [hunk 4]
```diff
@@ -74,6 +74,48 @@ func NewCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) *CidrSet {
...
+// len64 returns the minimum number of bits required to represent x; the result is 0 for x == 0.
+func len64(x uint64) (n int) {
```
[the remark this answers is not in the document]
Comment (aanm) [reply pkg/controller/node/ipam/cidrset/cidr_set.go]: It's the same name from golang's lib. I just copy-paste

**Pull request merged**